image/remote.go

```diff
@@ -190,6 +191,10 @@ func (r *remote) TopLayer() (string, error) {
 	return hex.String(), nil
 }
 
+func (r *remote) GetLayer(string) (io.ReadCloser, error) {
+	panic("not implemented")
```

Comment:
Considering returning an error here instead of panicking. (Interfaces are evaluated at runtime, so may not always be a programmer error.)

Reply:
We'd like to leave the panics because we do think that although interfaces are evaluated at runtime, using the remote image in this way is strictly programmer error. That begs another discussion on the Image interface (it could be a more limited interface), and that would fail at compile time. But, that seems like a separate refactor.

Reply:
My thinking is:
1. This is not the pattern that I've observed elsewhere (e.g., in ggcr: https://github.com/google/go-containerregistry/blob/52aa2729ea7597c8d5f7d6a977fa5cab9fc3a77a/pkg/v1/stream/layer.go#L78)
2. It seems conceivable that a combination of user choices (e.g., remote + get layer) would rely on this method to inform the user that the combination is invalid. A panic is much less friendly in that case.
3. `lifecycle/image` is importable from outside of the lifecycle module, and it's generally bad practice in Go to panic across package boundaries. An external library could recover from this case (e.g., by propagating an error that the user selection was invalid).

(I'm not going to block merging the PR on consensus here, just providing feedback. 😄 )